dataframe-jdbc/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/readJdbc.kt

```diff
@@ -322,19 +347,119 @@ public fun Connection.readDataFrame(
+        "--", // Single-line comments
+        "/*", // Start of multi-line comments
+        "*/", // End of multi-line comments
+        "DROP",
+        "DELETE",
+        "INSERT",
+        "UPDATE",
+        "EXEC",
+        "EXECUTE",
+        "CREATE",
+        "ALTER",
+        "GRANT",
+        "REVOKE",
+        "MERGE",
+    )
+
+    for (pattern in forbiddenPatterns) {
+        if (input.contains(pattern)) {
+            logger.error {
+                "Validation failed: The input contains a forbidden element '$pattern'. " +
+                    "Please review the input: '$input'."
+            }
+            return true
+        }
+    }
+    return false
+}
+
+/**
+ * Validates if the SQL query is safe and starts with SELECT.
+ * Ensures proper syntax structure, checks for balanced quotes, and disallows dangerous commands or patterns.
+ */
+private fun isValidSqlQuery(sqlQuery: String): Boolean {
     val normalizedSqlQuery = sqlQuery.trim().uppercase()
 
-    return normalizedSqlQuery.startsWith(START_OF_READ_SQL_QUERY) &&
-        !normalizedSqlQuery.contains(MULTIPLE_SQL_QUERY_SEPARATOR)
+    // Log the query being validated
+    logger.warn { "Validating SQL query: '$sqlQuery'" }
+
+    // Ensure the query starts with "SELECT"
+    if (!normalizedSqlQuery.startsWith("SELECT")) {
+        logger.error { "Validation failed: The SQL query must start with 'SELECT'. Given query: '$sqlQuery'." }
+        return false
+    }
+
+    // Validate against forbidden patterns
+    if (containsForbiddenPatterns(normalizedSqlQuery)) {
+        return false
+    }
+
+    // Check if there are balanced quotes (single and double)
+    val singleQuotes = sqlQuery.count { it == '\'' }
+    val doubleQuotes = sqlQuery.count { it == '"' }
+    if (singleQuotes % 2 != 0) {
+        logger.error {
+            "Validation failed: Unbalanced single quotes in the SQL query. " +
+                "Please correct the query: '$sqlQuery'."
+        }
+        return false
+    }
+    if (doubleQuotes % 2 != 0) {
+        logger.error {
+            "Validation failed: Unbalanced double quotes in the SQL query. " +
+                "Please correct the query: '$sqlQuery'."
+        }
+        return false
+    }
+
+    logger.warn { "SQL query validation succeeded for query: '$sqlQuery'." }
+    return true
+}
+
+/**
+ * Validates if the given SQL table name is safe and logs any validation violations.
+ */
+private fun isValidTableName(tableName: String): Boolean {
+    val normalizedTableName = tableName.trim().uppercase()
+
+    // Log the table name being validated
+    logger.warn { "Validating SQL table name: '$tableName'" }
+
+    // Validate against forbidden patterns
+    if (containsForbiddenPatterns(normalizedTableName)) {
+        return false
+    }
+
+    // Validate the table name structure: letters, numbers, underscores, and dots are allowed
+    val tableNameRegex = Regex("^[A-Z0-9_]+(\\.[A-Z0-9_]+)*$")
```

Comment:
regex can be a const somewhere

Reply:
only uppercase letters are allowed? Oh I see, normalizedTableName is made uppercase, got it

Reply:
btw, do we specifically only allow A-Z? So, not Æ, Ø, Å, etc? Otherwise, it might make more sense to replace `[A-Z0-9_]` by unicode category with `[\p{L}\p{N}_]` (https://www.regular-expressions.info/unicode.html). If SQL has a specific ISO standard for these names, let's stick to that instead.

Reply:
Agree about possibility to use other name, it seems inconsistent with other DataFrame parts, from other hand, I though that this behavior could be covered with disabling strictValidation

Reply:
Hmm yeah probably, but it seems unnecessarily strict to me. If you're from a country that uses a different alphabet you will always have to turn off `strictValidation` in this case.

Reply:
As I said I agree, we're supporting different alphabets and symbols in other stuff, better to be aligned with that, I consider this and add test cases (honestly never named in Cyrillic any table in my life)

Reply:
Now, Unicode is supported and test is added